File: mainActivity/src/main/java/com/sierrawireless/avphone/task/AsyncTaskFactory.java
package com.sierrawireless.avphone.task;

import android.content.Context;
import android.os.AsyncTask;

import net.airvantage.model.AvError;
import net.airvantage.utils.AirVantageClient;

public class AsyncTaskFactory implements IAsyncTaskFactory {

    private Context context;

    public AsyncTaskFactory(Context context) {
        this.context = context;
    }
    
    public SyncWithAvTask syncAvTask(String serverHost, String token) {
        
        AirVantageClient avClient = new AirVantageClient(serverHost, token);

        IApplicationClient appClient = new ApplicationClient(avClient);
        ISystemClient systemClient = new SystemClient(avClient);
        IAlertRuleClient alertRuleClient = new AlertRuleClient(avClient);
        IUserClient userClient = new UserClient(avClient);

        return new ProgressSyncWithAvTask(appClient, systemClient, alertRuleClient, userClient, context);
    }

    public GetUserTask getUserTak(String serverHost, String token) {

        AirVantageClient avClient = new AirVantageClient(serverHost, token);

        IUserClient userClient = new UserClient(avClient);

        return new ProgressGetUserTask(userClient, context);
    }

    public DeleteSystemTask deleteSystemTak(String serverHost, String token) {

        AirVantageClient avClient = new AirVantageClient(serverHost, token);
        ISystemClient systemClient = new SystemClient(avClient);

        IUserClient userClient = new UserClient(avClient);

        return new ProgressDeleteSystemTask(systemClient, userClient, context);
    }


    public AsyncTask<String, Integer, AvError> logoutTask(String serverHost, String token) {
        
        AirVantageClient client = new AirVantageClient(serverHost, token);

        return new LogoutTask(client);
    }
    
}

File: mainActivity/src/main/java/com/sierrawireless/avphone/task/AsyncTaskFactory.kt
package com.sierrawireless.avphone.task

import android.content.Context
import android.os.AsyncTask

import net.airvantage.model.AvError
import net.airvantage.utils.AirVantageClient

class AsyncTaskFactory(private val context: Context) : IAsyncTaskFactory {

    override fun syncAvTask(serverHost: String, token: String): SyncWithAvTask {

        val avClient = AirVantageClient(serverHost, token)

        val appClient = ApplicationClient(avClient)
        val systemClient = SystemClient(avClient)
        val alertRuleClient = AlertRuleClient(avClient)
        val userClient = UserClient(avClient)

        return ProgressSyncWithAvTask(appClient, systemClient, alertRuleClient, userClient, context)
    }

    override fun getUserTak(serverHost: String, token: String): GetUserTask {

        val avClient = AirVantageClient(serverHost, token)

        val userClient = UserClient(avClient)

        return ProgressGetUserTask(userClient, context)
    }

    override fun deleteSystemTak(serverHost: String, token: String): DeleteSystemTask {

        val avClient = AirVantageClient(serverHost, token)
        val systemClient = SystemClient(avClient)

        val userClient = UserClient(avClient)

        return ProgressDeleteSystemTask(systemClient, userClient, context)
    }


    override fun logoutTask(serverHost: String, token: String): AsyncTask<String, Int, AvError> {

        val client = AirVantageClient(serverHost, token)

        return LogoutTask(client)
    }
}
